### DISCUSSION TITLE
queue_connection sync vs database

### DISCUSSION THREAD

**Discussion by Muneeb-Ahmed-Khan** [spatie/laravel-stripe-webhooks#83] [Q&A]
Hi.
Thanks for reading my comment.
I have been using the wonderful package in my laravel project. currently the project is in development phase.
I am having a big confusion about (queue_connection sync vs database), I have read in multiple places that **sync** runs the Job on main thread and is not recommended for production.

Now, Suppose that the project is on production server with **queue_connection=sync** and 100 customers performs some action and generate 300 events. The project has subscribed to those events, now when all those events come and we have **sync**, most of them would be timed out and failed in stripe dashboard because the main thread would be blocked for couple of events and most of them would wait and in meantime the request would time out.
Am I right about this ? 

What should be done in order to  avoid this behavior and be able to entertain 
- multiple webhook events even though one event take ~10 second
- Giving users the access to website as well so they can use the website normally.

P.S : This is my first time opening issue or comment. I apologize in advance for my bad English.

**Comment by Hamaad641** [comment]
use redis and try to dsipatch your events on a specified queue and handle queue logic inside horizon.php like how many retries it would take or how many process it would run at a time
